README.md

```diff
@@ -10,82 +10,180 @@
 </p>
 
-<p align="center">
-  <a href="#key-features">Key Features</a> •
-  <a href="#build">Build</a> •
-  <a href="#run">Run</a> •
-  <a href="#documentation">Documentation</a> •
-  <a href="#related">Related</a> •
-  <a href="#license">License</a>
-</p>
+<details open>
+  
+  <summary><h2>Table of Contents</h2></summary>
+
+* [Key Features](#key-features)
+* [Quick Start](#quick-start)
+* [Build Options](#build-options)
+* [Using kvssink](#using-kvssink)
+* [Troubleshooting](#troubleshooting)
+* [Development](#development)
+* [Related](#related)
+* [License](#license)
+  
+</details>
+
+<br>
 
 ## Key Features
-* C++ SDK
+* C++ SDK with sample programs
 * GStreamer Plugin (kvssink)
-* JNI
+* Java Native Interface (JNI)
 
-Amazon Kinesis Video Streams Producer SDK for C/C++ makes it easy to build an on-device application that securely connects to a video stream, and reliably publishes video and other media data to Kinesis Video Streams. It takes care of all the underlying tasks required to package the frames and fragments generated by the device's media pipeline. The SDK also handles stream creation, token rotation for secure and uninterrupted streaming, processing acknowledgements returned by Kinesis Video Streams, and other tasks.
+Amazon Kinesis Video Streams Producer SDK for C/C++ makes it easy to build an on-device application that securely connects to a video stream, and reliably publishes video and other media data to Kinesis Video Streams (KVS). It takes care of all the underlying tasks required to package the frames and fragments generated by the device's media pipeline. The SDK also handles stream creation, token rotation for secure and uninterrupted streaming, processing acknowledgements returned by Kinesis Video Streams, and other tasks. The SDK includes a GStreamer plugin to stream directly to a custom GStreamer element kvssink, and includes a JNI to for use with Java applicatoins and the [KVS Java Producer SDK](https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-java).
 
-## Build
-### Download
-To download run the following command:
+<br>
+
+For more info on KVS please see [AWS KVS Documentation](https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/what-is-kinesis-video.html) and [Amazon Kinesis Video Streams FAQs](https://aws.amazon.com/kinesis/video-streams/faqs/).
+
+<br>
 
+## Quick Start
+### Required Tools
+The following packages are required to build the SDK libraries. Using a package manager such as _Homebrew_ (Mac), _APT_ (Linux), and _Chocolatey_ (Windows) is the prefered method of installation.
+* C++ Compiler (GNU or Clang recommended)
+* `git`
+* `CMake`
+* `pkg-config` for _Mac_/_Linux_, `pkgconfiglite` for _Windows_
+* `m4`
+* _Windows_ only: `nasm` and `strawberryperl`
+
```

Comment:
Suggest adding the installation commands for these.
```
sudo apt-get update
sudo apt-get install git cmake ...
```

Reply:
This seems like it would be unnecessarily verbose since these are very common packages which are likely already present on users' machines, but I'll add it and see how we feel about it.

Reply:
On second thought, it definitely would be to verbose to have all these commands for all 3 OSs, but it should fit well if we restructure the QuickStart section to be 3 separate sections for each OS. Will leave as is for now.

Reply:
> This seems like it would be unnecessarily verbose since these are very common packages which are likely already present on users' machines, but I'll add it and see how we feel about it.

In the case that it's not or if I want to verify for myself that it is installed? This information seems quite important to have.

Reply:
Added the installation steps 👍